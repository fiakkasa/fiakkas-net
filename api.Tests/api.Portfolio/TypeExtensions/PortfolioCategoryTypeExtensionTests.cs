using api.Portfolio.DataLoaders;
using api.Portfolio.Interfaces;
using api.Portfolio.Models;
using api.Shared.Interfaces;

namespace api.Portfolio.TypeExtensions;

public class PortfolioCategoryTypeExtensionTests
{
    [Fact]
    public async Task GetPortfolioItems_Should_Return_Data()
    {
        var items = new[]
        {
            new PortfolioItem
            {
                Id = Guid.Empty,
                CreatedAt = new (2024, 1, 1, 0, 0, 0, TimeSpan.Zero),
                UpdatedAt = null,
                Version = 1,
                Title = "Title",
                Href = new Uri("/test", UriKind.Relative),
                TechnologyIds = [Guid.Empty],
                CustomerId = Guid.Empty
            }
        };
        var dataRepository = Substitute.For<IDataRepository<IPortfolioItem>>();
        dataRepository
            .GetGroupedBatch(
                Arg.Any<IReadOnlyList<Guid>>(),
                Arg.Any<Func<IPortfolioItem, Guid>>(),
                Arg.Any<Func<IPortfolioItem, PortfolioItem>>(),
                Arg.Any<CancellationToken>()
            )
            .Returns(items.ToLookup(x => x.Id));
        var dataLoader = new PortfolioItemByPortfolioCategoryIdGroupDataLoader(
            dataRepository,
            AutoBatchScheduler.Default
        );
        var teut = new PortfolioCategoryTypeExtension();

        var result = await teut.GetPortfolioItems(
            new PortfolioCategory { Id = Guid.Empty },
            dataLoader,
            CancellationToken.None
        );

        result.Should().NotBeEmpty();
        result.MatchSnapshot();
    }
}
