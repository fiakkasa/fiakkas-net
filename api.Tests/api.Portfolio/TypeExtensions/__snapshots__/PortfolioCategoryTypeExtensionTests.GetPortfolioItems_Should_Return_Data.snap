[
  {
    "Year": 0,
    "CategoryId": "00000000-0000-0000-0000-000000000000",
    "Ordinal": 0,
    "Title": "Title",
    "Href": "/test",
    "TechnologyIds": [
      "00000000-0000-0000-0000-000000000000"
    ],
    "CustomerId": "00000000-0000-0000-0000-000000000000",
    "Id": "00000000-0000-0000-0000-000000000000",
    "CreatedAt": "2024-01-01T00:00:00+00:00",
    "UpdatedAt": null,
    "Version": 1
  }
]
